pkg/handler/loki.go

```diff
@@ -127,3 +68,79 @@ func executeLokiQuery(flowsURL string, lokiClient httpclient.HTTPClient) ([]byte
 	}
 	return resp, http.StatusOK, nil
 }
+
+func fetchParallel(lokiClient httpclient.HTTPClient, queries []string) ([]byte, int, error) {
+	// Run queries in parallel, then aggregate them
+	resChan := make(chan model.QueryResponse, len(queries))
+	errChan := make(chan errorWithCode, len(queries))
+	var wg sync.WaitGroup
+	wg.Add(len(queries))
+
+	for _, q := range queries {
+		go func(query string) {
+			defer wg.Done()
+			resp, code, err := executeLokiQuery(query, lokiClient)
+			if err != nil {
+				errChan <- errorWithCode{err: err, code: code}
+			} else {
+				var qr model.QueryResponse
+				err := json.Unmarshal(resp, &qr)
+				if err != nil {
+					errChan <- errorWithCode{err: err, code: http.StatusInternalServerError}
+				} else {
+					resChan <- qr
+				}
+			}
+		}(q)
+	}
+
+	wg.Wait()
+	close(resChan)
+	close(errChan)
+
+	for errWithCode := range errChan {
+		return nil, errWithCode.code, errWithCode.err
+	}
+
+	// Aggregate results
+	var aggregated model.QueryResponse
+	var aggStreams model.Streams
+	for r := range resChan {
+		if streams, ok := r.Data.Result.(model.Streams); ok {
+			if len(aggStreams) == 0 {
+				aggStreams = streams
+				aggregated = r
+			} else {
+				aggStreams = merge(aggStreams, streams)
+				aggregated.Data.Result = aggStreams
+			}
+		} else {
+			return nil, http.StatusInternalServerError, fmt.Errorf("loki returned an unexpected type: %T", r.Data.Result)
+		}
+	}
+
+	// Encode back to json
+	encoded, err := json.Marshal(aggregated)
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+
+	return encoded, http.StatusOK, nil
+}
+
+func merge(into, from model.Streams) model.Streams {
+	// TODO: o(n²), to optimize
```

Comment:
Should we use flowlogs pipeline unique hashkey here (or equivalent) and then create a simple map[string]model.Stream ?